perses/rjmc/atom_mapping.py

```diff
@@ -0,0 +1,1475 @@
+            # https://github.com/choderalab/perses/blob/412750c457712da1875c7beabfe88b2838f7f197/perses/rjmc/topology_proposal.py#L1123
+
+            old_oeatoms = { oeatom.GetIdx() : oeatom for oeatom in atom_mapping.old_mol.to_openeye().GetAtoms() }
+            new_oeatoms = { oeatom.GetIdx() : oeatom for oeatom in atom_mapping.new_mol.to_openeye().GetAtoms() }
+
+            # Generate filtered mappings
+            old_to_new_atom_map = atom_mapping.old_to_new_atom_map
+
+            mapped_atoms = {
+                old_index : new_index for old_index, new_index in old_to_new_atom_map.items()
+                }
+
+            mapped_aromatic_atoms = {
+                old_index : new_index for old_index, new_index in old_to_new_atom_map.items()
+                    if old_oeatoms[old_index].IsAromatic() and new_oeatoms[new_index].IsAromatic()
+                }
+
+            mapped_heavy_atoms = {
+                old_index : new_index for old_index, new_index in old_to_new_atom_map.items()
+                    if (old_oeatoms[old_index].GetAtomicNum()>1) and (new_oeatoms[new_index].GetAtomicNum()>1)
+                }
+
+            mapped_ring_atoms = {
+                old_index : new_index for old_index, new_index in old_to_new_atom_map.items()
+                    if old_oeatoms[old_index].IsInRing() and new_oeatoms[new_index].IsInRing()
+                }
+
+            # These weights are totally arbitrary
+            map_score = 1.0 * len(mapped_atoms) \
+                      + 0.8 * len(mapped_aromatic_atoms) \
+                      + 0.5 * len(mapped_heavy_atoms) \
+                      + 0.4 * len(mapped_ring_atoms)
+
+        return map_score
+
+    def generate_atom_mapping_from_positions(self, old_mol, new_mol):
+        """Generate an atom mapping derived entirely from atom position proximity.
+
+        The resulting map will be cleaned up by de-mapping hydrogens and rings as needed.
+
+        Parameters
+        ----------
+        old_mol : openff.toolkit.topology.Molecule or openeye.oechem.OEMol
+            The initial molecule for the transformation.
+        new_mol : openff.toolkit.topology.Molecule or openeye.oechem.OEMol
+            The final molecule for the transformation.
+
+        Returns
+        -------
+        atom_mapping : AtomMapping
+            The atom mapping determined from positions.
+            mapping[molB_index] = molA_index is the mapping of atoms from molA to molB that are geometrically close
+
+        Examples
+        --------
+        Derive atom mapping from positions:
+
+        >>> atom_mapper = AtomMapper()
+        >>> old_mol = Molecule.from_file('old_mol.sdf')
+        >>> new_mol = Molecule.from_file('new_mol.sdf')
+        >>> atom_mapping = atom_mapper.generate_atom_mapping_from_positions(old_mol, new_mol)
+
+        """
+        from openff.toolkit.topology import Molecule
+        distance_unit = unit.angstroms # distance unit for comparisons
```

Comment:
Are units always expected to be angstroms? I guess this works for the most part but maybe we could try to get the actual units from the molecule conformers themselves. Something like `old_mol.conformers[0].unit` (assuming they all have the same units as the first occurrence)

Reply:
The `distance_unit` here is arbitrary, but defined here to be consistent in how both the measured distances and absolute distance threshold are converted into unitless numbers. So the choice is totally arbitrary.